tools/codegen/main.go

```diff
@@ -27,21 +30,61 @@ func main() {
 		log.Fatalf("an error occurred when downloading Atlas Admin API spec: %v", err)
 	}
 
-	model, err := codespec.ToCodeSpecModel(specFilePath, configPath, resourceName)
-	if err != nil {
-		log.Fatalf("an error occurred while generating codespec.Model: %v", err)
+	{
```

Comment:
curious about the redundant curly brackets, added intentional to separate related statements? We can also consider extracting all the model serialization to separate function to keep main more concise